hummingbot/connector/exchange/k2/k2_exchange.py

```diff
@@ -0,0 +1,858 @@
+                                           tracked_order.executed_amount_quote,
+                                           tracked_order.fee_paid,
+                                           tracked_order.order_type))
+            self.stop_tracking_order(tracked_order.client_order_id)
+
+    async def _process_balance_message(self, balance_msg: Dict[str, Any]):
+        for asset_entry in balance_msg["data"]:
+            asset_name = asset_entry["coin"]
+            self._account_available_balances[asset_name] = Decimal(str(asset_entry["availablebalance"]))
+            self._account_balances[asset_name] = Decimal(str(asset_entry["balance"]))
+
+    async def get_open_orders(self) -> List[OpenOrder]:
+        result = await self._api_request(method="POST",
+                                         path_url=constants.GET_OPEN_ORDERS,
+                                         is_auth_required=True
+                                         )
+        ret_val = []
+        for order in result["data"]:
+            # TradeType is not provided by v1/Private/GetOpenOrders endpoint
+            # if order["type"] != "LIMIT":
+            #     raise Exception(f"Unsupported order type {order['type']}")
+            ret_val.append(
+                OpenOrder(
+                    client_order_id=order["orderid"],
+                    trading_pair=k2_utils.convert_from_exchange_trading_pair(order["symbol"]),
+                    price=Decimal(str(order["price"])),
+                    amount=Decimal(str(order["quantity"])),
+                    executed_amount=Decimal(str(order["quantity"])) - Decimal(str(order["leaveqty"])),
+                    status=constants.ORDER_STATUS[order["status"]],
+                    order_type=OrderType.LIMIT,
+                    is_buy=True if order["type"] == "Buy" else False,
+                    time=int(time.time() * 1e3),  # TODO: Check in with K2 for order creation ts, order update ts
+                    exchange_order_id=order["orderid"]  # TODO: Check in with K2 for unique order id
+                )
+            )
+        return ret_val
+
+    async def cancel_all(self, timeout_seconds: float):
+        """
+        Cancels all in-flight orders and waits for cancellation results.
+        Used by bot's top level stop and exit commands (cancelling outstanding orders on exit)
+        :param timeout_seconds: The timeout at which the operation will be canceled.
+        :returns List of CancellationResult which indicates whether each order is successfully cancelled.
+        """
+        if self._trading_pairs is None:
+            raise Exception("cancel_all can only be used when trading_pairs are specified.")
+        cancellation_results = []
+
+        for trading_pair in self._trading_pairs:
+            api_params = {
+                "symbol": k2_utils.convert_to_exchange_trading_pair(trading_pair)
+            }
+            try:
+                await self._api_request(method="POST",
```

Comment:
I think this would cancel all orders independently if they were or not created by hummingbot. It would be more suitable to have a batch cancel request with the orders living in in_flight_orders collection

Reply:
Updated `cancel_all()` to batch cancel orders being placed by the bot.